connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerTaskTest.java

```diff
@@ -78,155 +74,80 @@ public void tearDown() {
     public void standardStartup() {
         ConnectorTaskId taskId = new ConnectorTaskId("foo", 0);
 
-        WorkerTask workerTask = partialMockBuilder(WorkerTask.class)
-                .withConstructor(
-                        ConnectorTaskId.class,
-                        TaskStatus.Listener.class,
-                        TargetState.class,
-                        ClassLoader.class,
-                        ConnectMetrics.class,
-                        RetryWithToleranceOperator.class,
-                        Time.class,
-                        StatusBackingStore.class
-                )
-                .withArgs(taskId, statusListener, TargetState.STARTED, loader, metrics,
-                        retryWithToleranceOperator, Time.SYSTEM, statusBackingStore)
-                .addMockedMethod("initialize")
-                .addMockedMethod("initializeAndStart")
-                .addMockedMethod("execute")
-                .addMockedMethod("close")
-                .createStrictMock();
-
-        workerTask.initialize(TASK_CONFIG);
-        expectLastCall();
-
-        workerTask.initializeAndStart();
-        expectLastCall();
-
-        workerTask.execute();
-        expectLastCall();
-
-        statusListener.onStartup(taskId);
-        expectLastCall();
-
-        workerTask.close();
-        expectLastCall();
-
-        statusListener.onShutdown(taskId);
-        expectLastCall();
-
-        replay(workerTask);
+        WorkerTask workerTask = new TestWorkerTask(taskId, statusListener, TargetState.STARTED, loader, metrics,
+                retryWithToleranceOperator, Time.SYSTEM, statusBackingStore);
 
         workerTask.initialize(TASK_CONFIG);
         workerTask.run();
         workerTask.stop();
         workerTask.awaitStop(1000L);
 
-        verify(workerTask);
+        verify(statusListener).onStartup(eq(taskId));
+        verify(statusListener).onShutdown(eq(taskId));
```

Comment:
Nit: there's some inconsistency here in whether `eq(...)` is used to wrap verified arguments. I don't have a strong preference either way, but probably best to choose one and stick with one in order to avoid spreading FUD.